rasa/shared/core/flows/flow.py

```diff
@@ -582,6 +582,10 @@ def first(self) -> Optional[FlowStep]:
             return None
         return self.child_steps[0]
 
+    def fingerprint(self) -> str:
```

Comment:
should we make this a property? (maybe even a cached one given how often we are going to call it)